### DISCUSSION TITLE
HASPmota Why does a picture eat so much memory

### DISCUSSION THREAD

**Discussion by waltkind** [arendst/Tasmota#21100] [Q&A] [answered]
I use my Lanbon L8 as center for my home automation and also showing a picture of the door camera when someone is ringing.
The picture of the camera is converted to .png and scaled to 240 pixel width by my server. So the file is only 15kB.
Each time urlfetch imports a picture, the PSRfree is reduced by around 160 to 170kB.
I have limited the stored pictures to five and always delete the oldest. 
But the PSRFree can shrink from around 3.800kB at startup to around 2000kB. Then the system begins to become unresponsive and I have to restart it. Heap is constant around 140.

**Comment by barbudor** [comment]
Simple math
Your 15KB file is compressed
Assuming the file is 240 x 180 that is 43200 pixels
In most case each pixels requires 2 bytes (RGB 5:6:5 format) which is 84KB
May be there is another copy which would explain the 160-170KB

**Reply by waltkind** [reply]
Thank you for the explanation. I have no idea where the second copy should be.
Are the decompressed pictures kept in RAM when they are replaced by the next one?

**Comment by s-hadinger** [comment, marked as answer]
Yes PNG images are cached. This part of code had some regressions in LVGL9 and is being reworked by LVGL team

**Reply by waltkind** [reply]
Thank you. Now I understand.

**Reply by waltkind** [reply]
Is there a command to clear the cache?

**Reply by s-hadinger** [reply]
No unfortunately. I haven't dived in how the cache was implemented, so maybe there's an indirect way
